coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1020,7 +1020,25 @@ func (s *server) FailJob(ctx context.Context, failJob *proto.FailedJob) (*proto.
 			return nil, err
 		}
 
-		s.notifyWorkspaceBuildFailed(ctx, workspace, build)
+		manualBuild := build.Reason.Valid() && build.Reason == database.BuildReasonInitiator
+		if !manualBuild {
+			s.notifyWorkspaceAutoBuildFailed(ctx, workspace, build)
+		} else {
+			template, err := s.Database.GetTemplateByID(ctx, workspace.TemplateID)
+			if err != nil {
+				return nil, xerrors.Errorf("get template to notify manual build failed: %w", err)
+			}
+
+			owner, err := s.Database.GetUserByID(ctx, workspace.OwnerID)
+			if err != nil {
+				return nil, xerrors.Errorf("get owner to notify manual build failed: %w", err)
+			}
+
+			// Only notify the template creator if the build was initiated by someone else
+			if build.InitiatorID != template.CreatedBy {
```

Comment:
Why just the template creator, and not all template admins - for example?

Reply:
I followed the instructions described in issue https://github.com/coder/team-coconut/issues/16, and in my opinion, it makes sense. I believe only the template owner should receive build failure notifications from workspaces built from that template; otherwise, it could become too spammy.

>  manual build failure (receiver: owner)

Reply:
Many customers have 1-2 templates and 3-5 template administrators. We should send it to all template admins.